contracts/auth.ts

```diff
@@ -0,0 +1,72 @@
+/**
+ * Contract source: https://git.io/JOdz5
+ *
+ * Feel free to let us know via PR, if you find something broken in this
+ * file.
+ */
+
+import User from 'App/Models/User';
+
+declare module '@ioc:Adonis/Addons/Auth' {
```

Comment:
What is a contract?

Reply:
A contract is basically a way that describes exactly how your program should work within the framework and then checks those contracts such as auth.ts, events.ts at runtime. If something goes wrong, it pinpoints exactly what section of code was to blame with great descriptive error messages.

Reply:
Interessting! Again, what is your feeling of benefits over any other webframework? Express or what not?

Reply:
The most obvious benefit is, adonisjs lets me focus on building the application, so I don't have to focus much on building architecture around it as it comes with DB setup, mails, MVC, testing utilities, social authentication, you name it, and it is also well structured compared to expressjs and others.